synapse/cores/ram.py

```diff
@@ -46,6 +49,17 @@ def _sizeByLe(self, prop, valu, limit=None):
     def _rowsByLe(self, prop, valu, limit=None):
         return [ r for r in self.rowsbyprop.get(prop,()) if isint(r[2]) and r[2] <= valu ][:limit]
 
+    def _sizeByIn(self, prop, valu, limit=None):
+        lookup = set(valu)
```

Comment:
for small numbers of items, a tuple is a faster for "in" comparison than a set ( small number of iterative direct value comparisons is faster than doing the hash tree lookups for dict/set ).  Lets make this only "set-ify" the inbound if it's bigger than... say 12?